x-pack/test/detection_engine_api_integration/utils/delete_all_signals.ts

```diff
@@ -7,25 +7,37 @@
 
 import type SuperTest from 'supertest';
 import type { ToolingLog } from '@kbn/tooling-log';
+import type { Client } from '@elastic/elasticsearch';
 import { DETECTION_ENGINE_INDEX_URL } from '@kbn/security-solution-plugin/common/constants';
 import { countDownTest } from './count_down_test';
 
 /**
- * Deletes the signals index for use inside of afterEach blocks of tests
- * @param supertest The supertest client library
+ * Deletes all alerts from a given index or indices, defaults to `.alerts-security.alerts-default`
+ * For use inside of afterEach blocks of tests
  */
-export const deleteSignalsIndex = async (
+export const deleteAllSignals = async (
   supertest: SuperTest.SuperTest<SuperTest.Test>,
-  log: ToolingLog
+  log: ToolingLog,
+  es: Client,
+  index: string[] = ['.alerts-security.alerts-default']
 ): Promise<void> => {
   await countDownTest(
     async () => {
       await supertest.delete(DETECTION_ENGINE_INDEX_URL).set('kbn-xsrf', 'true').send();
+      await es.deleteByQuery({
+        index,
+        body: {
+          query: {
+            match_all: {},
+          },
+        },
+        refresh: true,
+      });
```

Comment:
If you go with option [1] we can keep this as-is, however with option [2] we will have to re-write this call to the delete by query api; I'm thinking something along these lines:

```suggestion
      await supertest.delete(DETECTION_ENGINE_INDEX_URL).set('kbn-xsrf', 'true').send();
      const allSpaces = await spaces.getAll();
      await Promise.all(
        allSpaces.map((space) =>
          es.deleteByQuery({
            index: `${index}${space.id}`,
            body: {
              query: {
                match_all: {},
              },
            },
            ignore_unavailable: true,
            refresh: true,
          })
        )
      );
```